File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/cpp/non-vul/code_before32.cpp
// Source: Row 129 in ./dataset/CVEfixes/Analysis/results/C++/df_cpp_all.xlsx

void Messageheader::Parser::checkHeaderspace(unsigned chars) const
{
if (headerdataPtr + chars >= header.rawdata + sizeof(header.rawdata))
{
    header.rawdata[sizeof(header.rawdata) - 1] = '\0';
    throw HttpError(HTTP_REQUEST_ENTITY_TOO_LARGE, "header too large");
}
}

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/python/non-vul/code_before32.py
# Source: Row 48 in ./dataset/CVEfixes/Analysis/results/Python/df_python_all.xlsx

def logout(request):
    """ Clears the session and logs the current user out. """
    request.user_logout()
    # FIXME(gabriel): we don't ship a view named splash
    return shortcuts.redirect('splash')